src/main/java/ch/uzh/ifi/hase/soprafs23/controller/GroupController.java

```diff
@@ -380,4 +383,32 @@ public void rejectJoinRequest(@PathVariable Long groupId, @RequestBody JoinReque
         joinRequestService.rejectJoinRequest(joinRequest);
     }
 
+    @PutMapping("/groups/{groupId}/ratings/{userId}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateRatings(@PathVariable Long groupId, @PathVariable Long userId,
+                                        @RequestBody List<IngredientPutDTO> ingredientPutDTOList,
+                                        HttpServletRequest request) {
+
+        Group group = groupService.getGroupById(groupId); // 404 - group not found
+
+        // check for authorization with request
+        List<Long> memberIds = groupService.getAllMemberIdsOfGroup(group);
+        Long tokenId = userService.getUseridByToken(request.getHeader("X-Token")); // 401 - not authorized
+        if(!memberIds.contains(tokenId)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, String.format("You are not a member of the group with id %d.", groupId));
+        }
+
+        // Convert the list of IngredientPutDTOs to a Map of ingredient IDs to ratings
+        Map<Long, String> ingredientRatings = ingredientPutDTOList.stream()
+                .collect(Collectors.toMap(IngredientPutDTO::getId, IngredientPutDTO::getUserRating));
+
+        // Pass the map of ingredient ratings to the service method
+        userService.updateIngredientRatings(groupId, userId, ingredientRatings);
+
+        // As last step calculate the final ingredientRating per Group and store it in ingredientRepo
+        groupService.calculateRatingPerGroup(groupId);
+
+        //TODO: then it depends on votingtype to get final result, so include API Call with voting type?
```

Comment:
I'd rather wrap the calculateRatingPerGroup with an if checking if the votingType of the group is MAYOTIYVOTING. atm the way our code is in the group creation the votingType is set to MAYORITYVOTE no matter what, so the check is not strictly speaking necessary imo, but I think passing along the votingtype with the call is unnecesary, because the backend can find out the votingType of the group

Reply:
thanks for the comment - I implemented it